src/context/auth.context.js

```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useEffect, useState } from "react";
+
+const User = createContext({ user: {} });
+
+const UserProvider = ({ children }) => {
+  const token = localStorage.getItem("token");
+  const [user, setUser] = useState(null);
+
+  useEffect(() => {
+    setUser(token);
+  }, [token]);
```

Comment:
Maybe you can add a check here whether your token is empty or not. But its a personal suggestion tough. I hope that in future you remove that token key from local storage after logout so that this logic would work perfectly fine.

Reply:
Yes taken care of this in Navbar component. Deleting the token post logout.